_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,652 @@
+---
+layout: default
+title: Search with k-NN filters
+nav_order: 10
+parent: k-NN
+has_children: false
+has_math: true
+---
+
+# Search for k-NN with filters
+Introduced 2.4
+{: .label .label-purple }
+
+You can create custom filters using Query DSL search options to refine your k-NN searches. You define the filter criteria within the `knn_vector` field's `filter` subsection in your query. You can use any of the OpenSearch Query DSL query types as a filter. This includes, but is not limited to these common query types: `term`, `range`, `regexp`, `wildcard`, and custom query types.  To include or exclude results, you specify Boolean query clauses. You also specify a query point with the `knn_vector` type and search for nearest neighbors that match your filter criteria.
+To run k-NN queries with a filter, the Lucene search engine and HSNW method are required.
+
+To learn more about how to use Query DSL Boolean query clauses, see [Boolean queries]({{site.url}}{{site.baseurl}}/opensearch/query-dsl/bool).
+{: .note }
+
+## How does a k-NN filter work?
+
+The OpenSearch k-NN plugin version 2.2 provided support for the Lucene engine to process k-NN searches. The Lucene engine provides a search that is based on the Hierarchical Navigable Small World (HSNW) algorithm to represent a multi-layered graph. The OpenSearch k-NN plugin version 2.4 is able to incorporate filters for searches based on Lucene 9.4.
+
+After a filter is applied to a set of documents to be searched, the algorithm decides whether to perform pre-filtering for an exact kNN search or modified post-filtering for approximate search. The approximate search with filtering guarantees the top number of closest vectors in result.
+
+Lucene also provides the capability to operate its `KnnVectorQuery` over a subset of documents. To learn more about Lucene’s new capability, see the [Apache Lucene Documentation](https://issues.apache.org/jira/browse/LUCENE-10382).
+
+### Filtered search performance
+
+Filtering that is tightly integrated with the Lucene HNSW algorithm implementation allows you to to apply k-NN searches more efficiently both in terms of relevancy of search results and performance. Consider that if you do an exact search on a large data set, the results are slow, and post-filtering does not guarantee the required number of results you specify for the `k` value.
+With this new capability, you can create an approximate k-NN search and apply filters, with the amount of results you need.
+
+The following workflow diagram shows how the HSNW algorithm decides which type of filtering to apply to a search based on the volume of documents, and number of `k` points in the index that you search with a filter. The variables shown in the diagram are described in the table below.
```

Comment:
As soon as you say, "The following workflow diagram..." I think you need to provide the diagram immediately after, rather than introduce a table before it.
1. diagram
2. key for variables.
Can the key be incorporated in the diagram or come right after it?

Reply:
yes. we added the descriptive table, but makes sense to put it below the diagram. thanks